File: app/src/main/java/com/chooloo/www/callmanager/ui/settings_fragment/SettingsMvpPresenter.java
package com.chooloo.www.callmanager.ui.settings_fragment;

import androidx.preference.Preference;

import com.chooloo.www.callmanager.ui.base.MvpPresenter;

public interface SettingsMvpPresenter<V extends SettingsMvpView> extends MvpPresenter<V> {
    void onRequestPermissionResult(int requestCode, int[] grantResults);

    void refresh();

    boolean onListPreferenceChange(Preference preference, Object newValue);

    boolean onSwitchPreferenceChange(Preference preference, Object newValue);

    boolean onCheckBoxPreferenceChange(Preference preference, Object newValue);
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/settings_fragment/SettingsMvpPresenter.kt
package com.chooloo.www.callmanager.ui.settings_fragment

import androidx.preference.Preference
import com.chooloo.www.callmanager.ui.base.MvpPresenter

interface SettingsMvpPresenter<V : SettingsMvpView> : MvpPresenter<V> {

    fun onRequestPermissionResult(requestCode: Int, grantResults: IntArray)

    fun onListPreferenceChange(preference: Preference, newValue: Any): Boolean
    fun onSwitchPreferenceChange(preference: Preference, newValue: Any): Boolean
    fun onCheckBoxPreferenceChange(preference: Preference, newValue: Any): Boolean

    fun refresh()
}